### ISSUE TITLE
Icons in menu_pages

### ISSUE THREAD

**Issue by gmt-it** [quellenform/t3x-iconpack#26]
Hello.

I have a question. Is there a way to display the icons in menu_pages for the pages inserted as records.

TIA

Guido

**Comment by stephankellermayr** [comment]
@gmt-it Unfortunately, I don't understand exactly what you mean. Can you explain this in more detail or give an example?

**Comment by gmt-it** [comment]
Hello Stephan.

I use menu_pages in the bootstrap package. Icons are stored for the pages to be displayed. These are not displayed in the Menu Pages. I have already tried to adjust MenuPages.typoscript and MenuPages.html, but I cannot get the icons to be displayed. {page.icon} in MenuPages.html is always empty. I have made a big mistake somewhere.

Bootstrap-package subnavigation:
![image](https://github.com/user-attachments/assets/06f92455-1c19-4371-8af7-6d957cf8cd70)

Bootstrap-package menu_pages:
![image](https://github.com/user-attachments/assets/50705bec-c74a-4417-b83b-3134afc380c7)

[MenuPages.typoscript.txt](https://github.com/user-attachments/files/17452397/MenuPages.typoscript.txt)
[MenuPages.html.txt](https://github.com/user-attachments/files/17452400/MenuPages.html.txt)

**Comment by stephankellermayr** [comment]
As far as I know, the TypoScript in this case is largely predefined by [TYPO3 MenuProcessor](https://github.com/TYPO3/typo3/blob/main/typo3/sysext/frontend/Classes/DataProcessing/MenuProcessor.php) and cannot be extended in this way.

However, since you already have the ViewHelper in your template anyway, you could insert the following code into the template:

```html
<ul>
	<f:for each="{menu}" as="page">
		<li>
			<i:icon iconfig="{page.data.page_icon}" preferredRenderTypes="svgInline,webfont" />
			<a href="{page.link}"{f:if(condition: page.target, then: ' target="{page.target}"')} title="{page.title}">{page.title}</a>
			<f:render section="Menu" arguments="{menu: page.children}" />
		</li>
	</f:for>
</ul>
```

**Comment by gmt-it** [comment]
Hell Stephan.

I've solved it now. The bootstrap package offers a separate content element for menu_pages. I've now added MenuPages.typoscript and MenuPages.html to my site set and adapted it to my needs. I have no idea why the bootstrap package doesn't provide for the display of icons in menu_pages.

I've attached the two files. Maybe someone else will look for them.

Thanks for your answer.

[MenuPages.html.txt](https://github.com/user-attachments/files/17462925/MenuPages.html.txt)
[MenuPages.typoscript.txt](https://github.com/user-attachments/files/17462934/MenuPages.typoscript.txt)

![image](https://github.com/user-attachments/assets/2b68506e-6584-4f8a-8a1d-7a491372444f)
